src/coreclr/src/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.cs

```diff
@@ -1114,7 +1115,7 @@ public virtual void MarkLabel(Label loc)
         #endregion
 
         #region IL Macros
-        public virtual void ThrowException(Type excType)
+        public virtual void ThrowException([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] Type excType)
```

Comment:
Shouldn't this be `DynamicallyAccessedMemberTypes.DefaultConstructor`? The code is calling `excType.GetConstructor(Type.EmptyTypes)`, which will only pick the default constructor.

Reply:
Linker is not smart enough to figure out the `Type.EmptyTypes` part so this warns. Maybe it should, along with `Array.Empty`.

Reply:
I've opened https://github.com/mono/linker/issues/1222 for this.